src/core/core.interaction.js

```diff
@@ -41,39 +41,38 @@ function evaluateAllVisibleItems(chart, handler) {
 	}
 }
 
+function useBinarySearch(axis, controller, _sorted, intersect) {
+	const iScale = controller._cachedMeta.iScale;
+	return iScale && axis === iScale.axis && _sorted &&
+		(!intersect || controller._elementsShareOptions);
```

Comment:
I see we check if there are shared options. But do we also need to get the size of the elements somewhere?

Reply:
Fixed